MAINT: Fix convergence message for remez

maniteja123 opened the pull request:
Correctly print the error message so as to describe the failure of convergence. Removes the "Design may still be correct" from the error message. Fixes #4569.

rgommers commented on scipy/signal/sigtoolsmodule.c:
```diff
@@ -1242,9 +1242,7 @@ static PyObject *sigtools_remez(PyObject *NPY_UNUSED(dummy), PyObject *args) {
-            sprintf(mystr, "Failure to converge after %d iterations.\n      " \
-                           "Design may still be correct.",
-                    maxiter);
+            printf(mystr, "Failure to converge after %d iterations.\n", maxiter);
```
the change from ``sprintf`` to ``printf`` isn't right. the message isn't printed to stdout but stored in ``mystr`` and used in the line below

maniteja123 commented:
Sorry, I had accidentally changed sprintf to printf. Fixed it now. Thanks @rgommers .

rgommers commented:
Can you squash the two commits and remove ``[ci skip]`` from the commit message? This is executable code, so the tests should be run. ``[ci skip]`` is really for when you touch docstrings or ``.txt`` files.

maniteja123 commented:
Oops, I didn't think about that since it was just a change in error message, sorry about that. I squashed the two commits and removed ``[skip ci]`` from the commit message. I also renamed the message to ``MAINT:...`` instead of ``DOC:...``.

rgommers commented:
Thanks, looks good now. ``MAINT`` is right. This can be merged when the testrun has completed.

rgommers commented:
Merged, thanks @maniteja123
